SQF/dayz_server/compile/server_handleSafeGear.sqf

```diff
@@ -104,27 +111,49 @@ switch (_status) do {
 	};
 };
 
-_type = switch _type do {
-    case "VaultStorage";
-    case "VaultStorageLocked": {
-		_lockCode = _charID;
-		"Safe"
-	};
-    case "LockboxStorage";
-    case "LockboxStorageLocked": {
-		_lockColor = "";
-		_lockCode =  parseNumber _charID;
-		_lockCode = _lockCode - 10000;
-		if (_lockCode <= 99) then { _lockColor = "Red"; };
-		if (_lockCode >= 100 && _lockCode <= 199) then { _lockColor = "Green"; _lockCode = _lockCode - 100; };
-		if (_lockCode >= 200) then { _lockColor = "Blue"; _lockCode = _lockCode - 200; };
-		if (_lockCode <= 9) then { _lockCode = format["0%1", _lockCode]; };
-		_lockCode = format ["%1%2",_lockColor,_lockCode];
-		"LockBox"
+_fnc_lockCode = {
+	private ["_color","_code"];
+
+	if (_this == "") exitWith {0};
+	_color = "";
+	_code = if (typeName _this == "STRING") then {parseNumber _this} else {_this};
+	_code = _code - 10000;
+	
+	if (_code <= 99) then {_color = localize "STR_TEAM_RED";};
+	if (_code >= 100 && _code <= 199) then {_color = localize "STR_TEAM_GREEN"; _code = _code - 100;};
+	if (_code >= 200) then {_color = localize "STR_TEAM_BLUE"; _code = _code - 200;};
+	if (_code <= 9) then {_code = format["0%1", _code];};
+	_code = format ["%1%2",_color,_code];
+	
+	_code
+};
+
+if (_status < 4) then {
+	_type = switch _type do {
+		case "VaultStorage";
+		case "VaultStorageLocked": {
+			_lockCode = _charID;
+			"Safe"
+		};
+		case "LockboxStorage";
+		case "LockboxStorageLocked": {
+			_lockCode = _charID call _fnc_lockCode;
+			if (_status == 3) then {_suppliedCode = _suppliedCode call _fnc_lockCode;};
+			"LockBox"
+		};
 	};
 };
 
-diag_log format["%1 (%2) %3 %4 with code: %5 @%6 (%7)",_name,_playerUID,_statusText,_type,_lockCode,mapGridPosition _pos,_pos];
+_lockCode = if (_type in DZE_DoorsLocked) then {_charID} else {_lockCode};
```

Comment:
You should change line 17 to `_lockCode = _charID;` then this line isn't needed and also line 135 wouldn't be needed anymore either.